gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -68,6 +70,76 @@ case class TransformPreOverrides(supportAdaptive: Boolean) extends Rule[SparkPla
     replaceWithTransformerPlan(project)
   }
 
+  /**
+   * If there are expressions (not field reference) in the partitioning's children, add a projection
+   * before shuffle exchange and make a new partitioning with the old expressions replaced by the
+   * result columns from the projection.
+   * */
+  private def addProjectionForShuffleExchange(plan: ShuffleExchangeExec): (Int, Partitioning,
```

Comment:
A little confusing to me. Is this function used to insert a projection before Shuffle? Looks simimar to below implementation. Maybe we can unify them into one?

https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala#L48-L69

cc @jinchengchenghh

Reply:
Need backend to support

Reply:
Velox backend also use computed hash value in HashPartitioning. Can you explain the difference with https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala#L48-L69 ?

Reply:
Under consideration, not compute the hash code for the whole keys, but let the backend compute the expressions idependently